From: Jason Rhinelander <jason@imaginary.ca>
Date: Wed, 18 May 2022 12:47:59 -0300
Subject: Change default DNS to 127.0.0.1:953

This is a needed workaround for a systemd resolved bug 23010, until
0.9.10 comes out (which will have a better fix to listen on both
127.3.2.1 + 127.0.0.1:random-high-port).
---
 llarp/config/config.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/llarp/config/config.cpp b/llarp/config/config.cpp
index 6bff961..a01f560 100644
--- a/llarp/config/config.cpp
+++ b/llarp/config/config.cpp
@@ -760,7 +760,9 @@ namespace llarp
     // can bind to other 127.* IPs to avoid conflicting with something else that may be listening on
     // 127.0.0.1:53.
 #ifdef __linux__
-    constexpr Default DefaultDNSBind{"127.3.2.1:53"};
+    // Workaround for systemd bug #23010 that breaks 127.0.0.0/8 resolvers on anything other than
+    // 127.0.0.1.  (This is temporary, until lokinet PR #1888 is merged in 0.9.10).
+    constexpr Default DefaultDNSBind{"127.0.0.1:953"};
 #else
     constexpr Default DefaultDNSBind{"127.0.0.1:53"};
 #endif
